frontend/app/services/personal-information-service.server.ts

```diff
@@ -247,49 +59,36 @@ function createPersonalInformationService() {
     throw new Error(`Failed to fetch data. address: ${response.status}, Status Text: ${response.statusText}`);
   }
 
-  function toPersonalInformation(personalInformationApiResponse: PersonalInformationApiResponse): PersonalInfo {
-    const addressList = personalInformationApiResponse.BenefitApplication.Applicant?.PersonContactInformation.at(0)?.Address;
-    const homeAddressList = addressList?.filter((address) => address.AddressCategoryCode.ReferenceDataName === 'Home');
-    const mailingAddressList = addressList?.filter((address) => address.AddressCategoryCode.ReferenceDataName === 'Mailing');
+  async function updatePersonalInformation(personalSinId: string, currentPersonalInformation: PersonalInfo, newPersonalInformation: PersonalInfo) {
+    const currentPersonalInfoDB = toPersonalInformationDB(currentPersonalInformation);
+    const newPersonalInfoDB = toPersonalInformationDB(newPersonalInformation);
```

Comment:
Mapping to the Database entities should be done in the Power Platform API mocks. We should be mapping using `toPersonalInformationApiRequest(..)` instead in this service.